### PR TITLE
Add proper notice if trying to update ASF that is already up-to-date

### LINKED ISSUE #106: Improve notifications for Update
If ASF is up to date, it should say so, instead of displaying Error message, I guess.

https://i.imgur.com/s7bvYq7.png

### PR DESCRIPTION
**Pull request by Ryzhehvost**
Closes #106

### CODE REVIEW

File: src/components/NavigationBrand.vue [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
 				try {
 					this.$info(this.$t('update-initiated'));
-					const response = await this.$http.post('asf/update');
+					const response = await this.$http.post('asf/update').catch;
```
Comment (Aareksio) [inline src/components/NavigationBrand.vue]: `.catch` accepts a callback function, eg.
```js
somePromise.catch(err => {
  // do something with the error
});
```

File: src/components/NavigationBrand.vue [hunk 2]
```diff
@@ -62,6 +62,12 @@
 						await waitForRestart();
 						this.$success(this.$t('restart-complete'));
 						window.location.reload(true);
+					} else {
+						if (response.Result!==null) {
+							this.$info(this.$t('update-uptodate', { message: response.message }));
+							response.Success=true;
```
Comment (Aareksio) [inline src/components/NavigationBrand.vue]: There is no point in altering `response.Success` here.

File: src/components/NavigationBrand.vue (hunk 1 above)
Comment (Ryzhehvost) [reply src/components/NavigationBrand.vue]: Do I understand it right, that if I add empty callback, try..catch won't work, and I actually need to re-throw error from there (if I want it to be handled)?

File: src/i18n/locale/default.json [hunk 3]
```diff
@@ -121,6 +121,7 @@
 	"update-check": "Checking for new version...",
 	"update-complete": "Updated!",
 	"update-initiated": "Updating...",
+	"update-uptodate": "ASF is up-to-date,{message}",
```
Comment (MrBurrBurr) [inline src/i18n/locale/default.json]: Shouldn't there be a space after the comma?

File: src/components/NavigationBrand.vue [hunk 4]
```diff
@@ -53,7 +53,11 @@
 
 				try {
 					this.$info(this.$t('update-initiated'));
-					const response = await this.$http.post('asf/update').catch;
+					const response = await this.$http.post('asf/update').catch(err => {
+						if (err.resonse && err.response.Result===null) {
+							throw err;
+						} else throw err;
```
Comment (Aareksio) [inline src/components/NavigationBrand.vue]: I like how you throw error no matter the condition :)

File: src/components/NavigationBrand.vue (hunk 2 above)
Comment (Ryzhehvost) [reply src/components/NavigationBrand.vue]: 🐶

File: src/components/NavigationBrand.vue (hunk 1 above)
Comment (JustArchi) [reply src/components/NavigationBrand.vue]: > If ASF returns proper http codes

This API endpoint will always generate 200 OK based on existing code, but you should still have logic for things like HTTP timeouts or other hiccups that might happen regardless.

File: src/components/NavigationBrand.vue (hunk 4 above)
Comment (Ryzhehvost) [reply src/components/NavigationBrand.vue]: :dog:

**Pull request closed without merging**